### PR TITLE
Fix invalid calculation of chart dimensions caused collapsing

### LINKED ISSUE #7544: Row chart - data hidden in the "others" category
Hello everyone,
I am using the row chart to plot these data in decreasing order:
table
However, the generated plot aggregates data from second to last in a large "others" row, and it's not possible to drill down the row,
row chart
I think that the condition for using the "others" category should not be triggered when the relative difference between rows is so small, and, possibly, there should be the option to select the "others" threshold.

### LINKED ISSUE #14285: Row Bar Chart collapses when there is value 0 in the last record
```sql
with temp as (
    select 'a' col1, 25 col2
    union all 
    select 'b', 10
    union all 
    select 'c', 15
    union all 
    select 'd', 0
    union all
    select 'e', 30
    union all 
    select 'f', 35
)
select * from temp
--order by 2 desc
```
In the first image  the `order by` clause is commented out. In the second image the `order by` clause is being evaluated (therefore the record with the value 0 in its second column is placed in the last row).  
This is in addition to the issue #13769 which is apparent in the images.

raw

raw2

v .37.2

### PR DESCRIPTION
**Pull request by alxnddr**
Closes https://github.com/metabase/metabase/issues/14285

Fix invalid calculations of bounds height, row height.

### First value is `0`
```sql
select 'a' c1 , 0 c2
union all 
select 'b' c1 , 2 c2
```

**Before (Both browsers)**
Screen Shot 2021-06-22 at 20 52 49

**After (both browsers)**
Screen Shot 2021-06-22 at 18 54 05

### Last value is `0`
```sql
select 'a' c1 , 2 c2
union all 
select 'b' c1 , 0 c2
```

**Before (Firefox only)**

Screen Shot 2021-06-22 at 20 53 14

**After (both browsers)**
Screen Shot 2021-06-22 at 18 52 43

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
Does it make sense to add a check/guard for `null` values, like we did in here https://github.com/metabase/metabase/pull/16687#pullrequestreview-687902599?

**Comment by alxnddr** [comment]
@nemanjaglumac thanks for checking, it works with `null` the same way as with `0`

**Comment by nemanjaglumac** [comment]
What I'm saying is - would it make sense to expand the test to guard against that in the future while we're at it? Seems like a low hanging fruit.

**Comment by alxnddr** [comment]
@nemanjaglumac oh, I got it now. Yes, it makes sense. I'll add the test, thanks!

File: frontend/src/metabase/visualizations/lib/RowRenderer.js [hunk 1]
```diff
@@ -159,24 +158,15 @@ export default function rowRenderer(
...
+  const rowHeight = (boundsHeight - chart.gap()) / group.size();
+  const barHeight = rowHeight - chart.gap();
+  const cap = Math.max(1, Math.floor(boundsHeight / ROW_MAX_HEIGHT));
   chart.cap(cap);
 
-  // assume all bars are same height?
-  const barHeight = chart.select("g.row")[0][0].getBoundingClientRect().height;
   if (barHeight > ROW_MAX_HEIGHT) {
-    const reasolableMaxGap = containerHeight / 3;
+    const reasolableMaxGap = boundsHeight / 3;
```
Comment (daltojohnso) [inline frontend/src/metabase/visualizations/lib/RowRenderer.js:169]: i'm guessing "reasolable" is a type for `reasonable`?

Comment (alxnddr) [reply frontend/src/metabase/visualizations/lib/RowRenderer.js]: Good catch, fixed that

**Comment by alxnddr** [comment]
@nemanjaglumac I've added an additional check for `null` as well, thanks for pointing it out

**Review by nemanjaglumac — APPROVED** [review]
Checked it locally (since we can't run firefox in CI). All working great! :)
image

**Review by daltojohnso — APPROVED** [review]
lgtm

**Pull request merged**